gnomad_constraint/plots/generate_constraint_plots.R

```diff
@@ -243,3 +309,226 @@ for (version in names(datasets)) {
+  (.data$v2 == TRUE &
+    .data$v4 == TRUE &
+    .data$constraint_flags == "[]" &
+    .data$mane_select == "true" &
+    grepl("^ENST", transcript))
+)
+comparision_df <- comparision_df %>%
+  select(
+    all_of(
+      c(
+        "transcript",
+        "v2",
+        "v4",
+        "oe_lof_upper",
+        "lof.oe_ci.upper",
+        "lof_z",
+        "lof.z_score"
+      )
+    )
+  )
+
+# Pivot to longer df with metric name and value
+comparision_df <- comparision_df %>%
+  pivot_longer(
+    cols = c(oe_lof_upper, lof.oe_ci.upper, lof_z, lof.z_score),
+    names_to = "metric_name",
+    values_to = "value"
+  )
+
+# Pull out version (v4 metrics have a "." in metric name whereas v2 metrics do not)
+comparision_df <- comparision_df %>% mutate(
+  version = if_else(grepl("\\.", metric_name), "v4", "v2")
+)
+
+# Rename metrics in df to be the same for v2 and v4
+v2_to_v4_mapping <- c(
+  oe_lof_upper = "lof.oe_ci.upper",
+  lof_z = "lof.z_score"
+)
+comparision_df <- comparision_df %>%
+  mutate(metric_name = if_else(metric_name %in% names(v2_to_v4_mapping),
+    v2_to_v4_mapping[metric_name],
+    metric_name
+  ))
+
+comparision_df <- comparision_df %>%
+  mutate(metric_name = case_when(
+    .data$metric_name == "lof.oe_ci.upper" ~ "LOEUF",
+    .data$metric_name == "lof.z_score" ~ "pLoF Z-score",
+    TRUE ~ .data$metric_name # Keeps all other values as they are
+  ))
+
+comparision_df <- comparision_df %>% select(-v2, -v4)
+
+# Pivot to wider format
+comparision_df <- comparision_df %>% pivot_wider(
+  names_from = version, values_from = value
+)
+
+# Calculate correlations for each metric between v2 and v4
+comparision_df %>%
+  group_by(metric_name) %>%
+  summarize(
+    correlation = cor(.data$v2, .data$v4, method = "pearson", use = "complete.obs")
+  )
+
+comparsion_plot <- plot_metric_comparison(comparision_df)
+plot_path <- get_plot_path(
+  "v2_v4_compare_values",
+  output_basedir = output_basedir
+)
+
+ggsave(
+  comparsion_plot,
+  filename = plot_path,
+  dpi = 300,
+  width = 8,
+  height = 4,
+  units = "in"
+)
+
+
+####################################################################
+####################################################################
+# Plot observed vs expected values
+####################################################################
+####################################################################
+for (version in versions_to_plot) {
+  filtered_data <- filter_transcripts(
+    constraint_data,
+    version,
+    preferred_transcripts_only = TRUE,
+    enst_only = TRUE,
+    include_flagged_transcripts = FALSE
+  )
+  # Reformat column names if data is from v2
+  if (version == "v2") {
+    oe_data <- filtered_data %>% select(
+      all_of(
+        c(
+          "transcript",
+          "v2",
+          "exp_syn",
+          "obs_syn",
+          "exp_mis",
+          "obs_mis",
+          "exp_lof",
+          "obs_lof"
+        )
+      )
+    )
+    oe_data <- oe_data %>% rename(
```

Comment:
Let's decide on which we want, `_` or `.` and change one in the loading function, then just use that throughout